wurst/util/UnitUtils.wurst

```diff
@@ -0,0 +1,60 @@
+package UnitUtils
+import BuffObjEditing
+import ClosureTimers
+
+
+
+/*	This file contains various utility functions for Unit, 
+	which require enhanced functionality like closures and cannot be defined in Unit.wurst */
+
+@configurable public real FIELD_OF_VIEW_FRONT  = 230.00
+@configurable public real FIELD_OF_VIEW_BEHIND = 115.00  // Real FoV is: 180 - 2x (FIELD_OF_VIEW-90)	Using a value of 115 generates a FoV of 130   (natural FoV is about 135 degrees)
+
+/** Apply Buff to this unit for duration.*/
+@configurable public function unit.applyBuffTimed(buffTuple buffApplied, real duration)
+	this.addAbility(buffApplied.abilId)
+	this.setAbilityLevel(buffApplied.abilId, 1)
+	doAfter(duration) ->
+		if this != null
+			this.setAbilityLevel(buffApplied.abilId, 0)
+			this.removeAbility(buffApplied.abilId)
+			
+/** Increase Armor of a unit for a duration */
+@configurable public function unit.addArmorTimed(real armorAdd, real duration)
+	BlzSetUnitArmor(this, this.getArmor()+armorAdd)
+	doAfter(duration) ->
+		if this != null
+			BlzSetUnitArmor(this, this.getArmor()-armorAdd)
+			
+/** Increase Damage of a unit for a duration */		
+@configurable public function unit.addDamageTimed(int damageAdd, real duration)
+	BlzSetUnitBaseDamage(this, this.getBaseDamage(1)+damageAdd, 1)
+	doAfter(duration) ->
+		if this != null
+			BlzSetUnitBaseDamage(this, this.getBaseDamage(1)-damageAdd, 1)
+
+/** Returns true if this unit is behind target unit.
+    Uses configurable FIELD_OF_VIEW_BEHIND variable with default value of 115. */
+public function unit.isBehind(unit targetUnit) returns bool
+	let degreeDif = this.getFacingAngle().degrees() - targetUnit.getFacingAngle().degrees()
+	return -FIELD_OF_VIEW_BEHIND <= degreeDif and degreeDif <= FIELD_OF_VIEW_BEHIND
+
+/** Returns true if this unit is in front of target unit.
+	Uses configurable FIELD_OF_VIEW_FRONT variable with default value of 230. */
+public function unit.isInFront(unit targetUnit) returns bool
+	let degreeDif = this.getFacingAngle().degrees() - targetUnit.getFacingAngle().degrees()
```

Comment:
I don't think it makes sene to compare two units facing angle for `inFront`ness.

We really want to compare one unit's facing angle to another unit's position

Reply:
Yeah its totally bullshit, also is behind

Reply:
Removed these old functions

Reply:
Here's some code that will do it:

```
    public function angle.smallestDiff(angle other) returns angle
        return this.direction().dot(other.direction()).acos().asAngleRadians()

    public function angle.toVec2() returns vec2
        return vec2(0., 0.).polarOffset(this, 1.)

    public function angle.lt(angle other) returns bool
        return (this.radians() % (2. * PI)) < other.radians()

    public function angle.toString() returns string
        return (this.degrees() % 360.).toString(2)

    public function unit.isFacing(unit which, real maxRange, angle maxAng) returns bool
        let angBetween = this.getPos().angleTo(which.getPos())
        let distBetween = this.getPos().distanceToSq(which.getPos())
        return (angBetween.smallestDiff(this.getFacingAngle())).lt(maxAng) and distBetween < maxRange*maxRange
```

Reply:
Yeah that is still not what I was talking about, I have a bit more complex code, picking all units in front or behind or in specific direction of the unit / pos. While allowing some filter, and configurable for search pattern shape and sizes. Iam not 100% if that is a thing for lib as it is a bit more math

Reply:
That seems hardly different from this